syntax = "proto3";

package extraction;

service ExtractionService {
  rpc Extract (ExtractionRequest) returns (stream ExtractionResponse);
}

message ExtractionRequest {
  string path = 1;
}

message ExInRegion {
  string sequence = 1;
  string type = 2;
  int32 start = 3;
  int32 end = 4;
  optional string gene = 5;
  optional int32 strand = 6;
  optional string before = 7;
  optional string after = 8;
}

message CDSRegion {
  string sequence = 1;
  string type = 2;
  int32 start = 3;
  int32 end = 4;
  string gene = 5;
}

message ExtractionResponse {
  string sequence = 1;
  string accession = 2;
  string organism = 3;
  repeated CDSRegion cds = 4;
  repeated ExInRegion exin = 5;
}
